provider/clusterrosaclassic/cluster_rosa_classic_resource.go

```diff
@@ -995,6 +1027,27 @@ func (r *ClusterRosaClassicResource) Read(ctx context.Context, request resource.
 	response.Diagnostics.Append(diags...)
 }
 
+func validateAdminCredentialsChange(ctx context.Context, state,
+	plan *ClusterRosaClassicState) diag.Diagnostics {
+	diags := diag.Diagnostics{}
+	if state.AdminCredentials.IsNull() {
+		if common.BoolWithFalseDefault(plan.CreateAdminUser) {
+			diags.AddError(common.AssertionErrorSummaryMessage, fmt.Sprintf(common.AssertionErrorDetailsMessage, "create_admin_user", false, true))
+		}
```

Comment:
What happen if AdminCredentials object exists and user change CreateAdminUser flag from true to false?
I think this case is not covered here

So maybe its better just to compare the bool value between the plan and state?
Something like:
```
common.ValidateStateAndPlanEquals(state.CreateAdminUser, plan.CreateAdminUser, "create_admin_user", &diags)
```
It can even be outside from 'validateAdminCredentialsChange' function

Reply:
@nirarg 
in current logic, if AdminCredentials exists, the value of CreateAdminUser will be ignored.
So for the case `AdminCredentials object exists and user change CreateAdminUser flag from true to false`, the flow will enter the `else` section to compare the `AdminCredentials` value in state&plan
is this ok?

Reply:
But I think that if users change it we should tell them explicit that it shouldn't be changed, otherwise they might expect to different result.
Changing only this attribute can cause to "update" in the next apply

Reply:
fixed